modules/infinite-scroll/infinity.php

```diff
@@ -435,6 +439,11 @@ function action_template_redirect() {
 		if ( empty( $id ) )
 			return;
 
+		// AMP infinite scroll functionality will start on amp_load_hooks().
+		if ( Jetpack_AMP_Support::is_amp_request() ) {
```

Comment:
We should check if class_exists here as per previous instruction on some of the other issues we've worked on